File: app/src/main/java/com/iita/akilimo/Locales.java
package com.iita.akilimo;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;

public class Locales {
    private static final Locale LOCAL_TZ_SWA = new Locale("sw", "TZ");
    private static final Locale LOCAL_KE_SWA = new Locale("sw", "KE");
    private static final Locale LOCAL_NG_ENGLISH = new Locale("en", "NG");

    public static final List<Locale> APP_LOCALES = Arrays.asList(
            Locale.ENGLISH,
            Locales.LOCAL_TZ_SWA);
    public static final List<Locale> LOCALE_COUNTRIES = Arrays.asList(Locales.LOCAL_TZ_SWA, LOCAL_NG_ENGLISH);
}

File: app/src/main/java/com/akilimo/mobile/utils/Locales.kt
package com.akilimo.mobile.utils


import android.content.Context
import android.content.res.Configuration
import androidx.core.content.edit
import java.util.Locale

data class LanguageOption(val code: String, val displayName: String)

object Locales {
    private val LOCAL_TZ_SWA = Locale("sw", "TZ")
    private val LOCAL_NG_ENGLISH = Locale("en", "NG")
    private val LOCAL_RW_KINYARWANDA = Locale("rw", "RW")

    val LOCALE_COUNTRIES: List<Locale> =
        listOf(LOCAL_NG_ENGLISH, LOCAL_TZ_SWA, LOCAL_RW_KINYARWANDA)
}

@Suppress(
    "kotlin:S6291",
    "kotlin:S2068"
) // Unencrypted preferences (justified: no sensitive info stored)
object LanguageManager {
    private const val LANGUAGE_KEY = "language_key"
    private const val DEFAULT_LANGUAGE = "en"

    fun saveLanguage(context: Context, language: String) {
        val sharedPreferences = context.getSharedPreferences("akilimo-prefs", Context.MODE_PRIVATE)
        sharedPreferences.edit { putString(LANGUAGE_KEY, language) }
    }

    fun getLanguage(context: Context): String {
        val sharedPreferences = context.getSharedPreferences("akilimo-prefs", Context.MODE_PRIVATE)
        return sharedPreferences.getString(LANGUAGE_KEY, DEFAULT_LANGUAGE) ?: DEFAULT_LANGUAGE
    }

    fun setLocale(context: Context, language: String): Context {
        val locale = Locale(language)
        Locale.setDefault(locale)
        val config = Configuration(context.resources.configuration)
        config.setLocale(locale)
        return context.createConfigurationContext(config)
    }
}
